### PR TITLE
Add assertions to commands, group and flashcards folder

### LINKED ISSUE #54: Add assertions to Commands
DeleteCommand and ListCommand: https://github.com/AY1920S2-CS2113-T14-1/tp/issues/67
...

### LINKED ISSUE #55: Add assertions to flashcards folder
FlashcardList: https://github.com/AY1920S2-CS2113-T14-1/tp/issues/68

### LINKED ISSUE #56: Add assertions to group folder

### PR DESCRIPTION
**Pull request by g-lilian**

### CODE REVIEW

File: src/main/java/seedu/tp/commands/GroupCommand.java [hunk 1]
```diff
@@ -18,6 +18,8 @@
      * @param groupList    groupList where the groupCommand execute on
      */
     public GroupCommand(GroupFactory groupFactory, GroupList groupList) {
+        assert groupList != null : "Invalid null GroupList!";
```
Comment (Zhilin-Huang) [inline src/main/java/seedu/tp/commands/GroupCommand.java:22]: Consider adding another assertion for `GroupFactory `?

File: src/main/java/seedu/tp/commands/OtherFlashcardCommand.java [hunk 2]
```diff
@@ -21,6 +21,8 @@
      * @param flashcardFactory the flashcard factory to be used in the command
      */
     public OtherFlashcardCommand(FlashcardList flashcardList, FlashcardFactory flashcardFactory) {
```
Comment (Zhilin-Huang) [inline src/main/java/seedu/tp/commands/OtherFlashcardCommand.java:23]: Consider adding another assertion for `FlashcardFactory`?

File: src/main/java/seedu/tp/commands/PersonFlashcardCommand.java [hunk 3]
```diff
@@ -21,6 +21,8 @@
      * @param flashcardFactory the flashcard factory to be used in the command
      */
     public PersonFlashcardCommand(FlashcardList flashcardList, FlashcardFactory flashcardFactory) {
+        assert flashcardList != null : "Invalid null FlashcardList!";
```
Comment (Zhilin-Huang) [inline src/main/java/seedu/tp/commands/PersonFlashcardCommand.java:24]: Same here

File: src/main/java/seedu/tp/commands/PriorityCommand.java [hunk 4]
```diff
@@ -23,6 +23,8 @@
      * @param pl            priority level to set the flashcard to
      */
     public PriorityCommand(FlashcardList flashcardList, int index, Ui ui, Flashcard.PriorityLevel pl) {
```
Comment (Zhilin-Huang) [inline src/main/java/seedu/tp/commands/PriorityCommand.java:25]: Same here

File: src/main/java/seedu/tp/commands/ReviewedCommand.java [hunk 5]
```diff
@@ -21,6 +21,8 @@
      * @param ui            instance for user interaction
      */
     public ReviewedCommand(FlashcardList flashcardList, int index, Ui ui) {
+        assert flashcardList != null : "Invalid null FlashcardList!";
```
Comment (Zhilin-Huang) [inline src/main/java/seedu/tp/commands/ReviewedCommand.java:24]: Consider adding another assertion for `Ui`?

File: src/main/java/seedu/tp/commands/ShowCommand.java [hunk 6]
```diff
@@ -22,6 +22,8 @@
      * @param ui            instance for user interaction
      */
     public ShowCommand(FlashcardList flashcardList, int index, Ui ui) {
+        assert flashcardList != null : "Invalid null FlashcardList!";
```
Comment (Zhilin-Huang) [inline src/main/java/seedu/tp/commands/ShowCommand.java:25]: Same here

File: src/main/java/seedu/tp/commands/TimelineCommand.java [hunk 7]
```diff
@@ -3,11 +3,22 @@
...
     private Ui ui;
 
+    /**
+     * Constructor for TimelineCommand.
+     *
+     * @param flashcardList     list containing all flashcards
+     * @param ui                instance for user interaction
+     */
     public TimelineCommand(FlashcardList flashcardList, Ui ui) {
+        assert flashcardList != null : "Invalid null FlashcardList!";
```
Comment (Zhilin-Huang) [inline src/main/java/seedu/tp/commands/TimelineCommand.java:20]: Same here

File: src/main/java/seedu/tp/group/GroupFactory.java [hunk 8]
```diff
@@ -11,7 +11,15 @@
...
     private FlashcardList flashcardList;
 
+    /**
+     * Constructor for the GroupFactory.
+     *
+     * @param ui                instance for user interaction
+     * @param flashcardList     list containing all flashcards
+     */
     public GroupFactory(Ui ui, FlashcardList flashcardList) {
+        assert flashcardList != null : "Invalid null FlashcardList!";
```
Comment (Zhilin-Huang) [inline src/main/java/seedu/tp/group/GroupFactory.java:22]: Assertion for `Ui`?

**Review by Zhilin-Huang — CHANGES_REQUESTED** [review]
Add logs/ to .gitignore so that we don't push the logs to the remote repo

File: src/main/java/seedu/tp/commands/GroupCommand.java (hunk 1 above)
Comment (g-lilian) [reply src/main/java/seedu/tp/commands/GroupCommand.java:22]: Noted and updated, removed logs from the commit too, thanks!

Comment (g-lilian) [reply src/main/java/seedu/tp/commands/GroupCommand.java:22]: Hmm this is really strange, at first the code passes checkstyle but after I commit again, it then shows me the checkstyle failed message due to missing Javadoc comments. The checkstyle test seems to be unreliable.

Comment (Zhilin-Huang) [reply src/main/java/seedu/tp/commands/GroupCommand.java:22]: > Hmm this is really strange, at first the code passes checkstyle but after I commit again, it then shows me the checkstyle failed message due to missing Javadoc comments. The checkstyle test seems to be unreliable.

Yeah I noticed this also, I guess when the method is very short (just a few lines), it does not demand a JavaDoc comment, but when it gets longer a JavaDoc comment is required

**Review by Zhilin-Huang — APPROVED** [review]

**Review by quinnyyy — APPROVED** [review]

**Review by sliu107 — APPROVED** [review]

**Pull request merged**